art/attacks/evasion/fast_gradient.py

```diff
@@ -341,6 +341,11 @@ def _compute_perturbation(
         # Get gradient wrt loss; invert it if attack is targeted
         grad = self.estimator.loss_gradient(batch, batch_labels) * (1 - 2 * int(self.targeted))
 
+        # Check for NaN before normalisation an replace with 0
+        if np.isnan(grad).any():
+            logger.warning("Elements of the loss gradient are NaN and have been replaced with 0.0.")
+            grad = np.where(np.isnan(grad), np.zeros_like(grad), grad)
```

Comment:
`np.zeros_like(grad)` can just be 0.0 (it will be broadcast by the method).

Reply:
Good idea.